eng/scripts/Language-Settings.ps1

```diff
@@ -67,16 +80,55 @@ function Get-Version ($pkgPath)
             # finding where the version number are
             if ($content -match $go_version_regex)
             {
-                return $matches["version"]
+                return $matches["version"], $versionFile
             }
         }
         catch
         {
-            Write-Error "Error parsing version."
+            LogError "Error parsing version."
             Write-Error $_
         }
     }
 
-    Write-Host "Cannot find release version."
-    exit 1
+    LogWarning "Cannot find release version."
+    return $null
+}
+
+function Get-AllPackageInfoFromRepo ($serviceDirectory)
+{
+    $allPackageProps = @()
+    $searchPath = Join-Path $RepoRoot "sdk"
+    if ($serviceDirectory)
+    {
+        $searchPath = Join-Path $searchPath $serviceDirectory 
+    }
+
+    $pkgFiles = Get-ChildItem -Path $searchPath -Include "go.mod" -Recurse
+
+    foreach ($pkgFile in $pkgFiles)
+    {
+        $pkgPath = $pkgFile.DirectoryName
+
+        $pkgName, $serviceDir, $sdkType = Get-GoPackageId -pkgDirectory $pkgPath
+        $pkgVersion, $versionFile = Get-Version -pkgPath $pkgPath
+
+        if ($pkgName -and $pkgVersion)
+        {
+            $pkgProp = [PackageProps]::new($pkgName, $pkgVersion, $pkgPath, $serviceDir)
+            $pkgProp.IsNewSdk = $true
+            $pkgProp.SdkType = $sdkType
+            $pkgProp.VersionFile = $versionFile
+            $allPackageProps += $pkgProp
+        }
```

Comment:
Is it worth adding some logging or something when the package name and version are not valid but we found a mod file?